pkg/workflows/internal/v2/sdkimpl/runtime.go

```diff
@@ -11,44 +11,51 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/pb"
 )
 
-type CallCapabilityFn func(request *pb.CapabilityRequest) (id []byte, err error)
+type CallCapabilityFn func(request *pb.CapabilityRequest) error
 type AwaitCapabilitiesFn func(request *pb.AwaitCapabilitiesRequest, maxResponseSize uint64) (*pb.AwaitCapabilitiesResponse, error)
 
 type RuntimeBase struct {
-	ExecId          string
 	ConfigBytes     []byte
 	MaxResponseSize uint64
 	Call            CallCapabilityFn
 	Await           AwaitCapabilitiesFn
 	Writer          io.Writer
 
-	modeErr error
+	modeErr    error
+	Mode       pb.Mode
+	nextCallId int32
 }
 
 func (r *RuntimeBase) CallCapability(request *pb.CapabilityRequest) sdk.Promise[*pb.CapabilityResponse] {
+	if r.Mode == pb.Mode_DON {
+		r.nextCallId++
```

Comment:
Is this really safe? If a user writes a multithreaded workflow, is it guaranteed that a single-threaded WASM sandbox will always execute it in the same way?

Reply:
WASM is single threaded. If/when they allow multi-threading, you would need to enable it on the host. 

Adding locks alone wouldn't be sufficient if it wasn't, because if two threads raced and we couldn't guarantee order anymore and the ID wouldn't match.

Reply:
Go routines seem very deterministic in WASM, they don't yield unless you do it explicitly or wait on a channel that doesn't have anything.

Given we're seeding random and time, I don't know where languages would be able to get non-determinism for routine/thread selection. Point taken though, I'm open to other ideas.

Reply:
You're probably right about the sources of non-determinism. I not super worried  right now but it would be cool to explore how exactly this behaves one day... Maybe leave a comment with a warning for now?